### PR TITLE
[Windows] Remove accessibility root assumptions

### PR DESCRIPTION
**Pull request by loic-sharma**
Today, the root accessibility semantics node is guaranteed to have ID 0. In a multi-window world, each view will have its own semantics tree, but semantic node IDs will be globally unique. In other words, the semantics tree's root will no longer be guaranteed to have ID 0.

This change removes the "root ID is 0" assumption from the Windows embedder. Note that other embedders (macOS, iOS, Fuchsia, etc...) also need updates, which will be done in follow-up PRs.

Part of https://github.com/flutter/flutter/issues/119391

## Manual test

<details>
<summary>Manual test...</summary>

Apply the framework patch below, run an app, and verify the accessibility works as expected:

```diff
diff --git a/packages/flutter/lib/src/semantics/semantics.dart b/packages/flutter/lib/src/semantics/semantics.dart
index 969f822f5f..6431b3ed28 100644
--- a/packages/flutter/lib/src/semantics/semantics.dart
+++ b/packages/flutter/lib/src/semantics/semantics.dart
@@ -1650,7 +1650,7 @@ class SemanticsNode extends AbstractNode with DiagnosticableTreeMixin {
     this.key,
     VoidCallback? showOnScreen,
     required SemanticsOwner owner,
-  }) : _id = 0,
+  }) : _id = 123,
        _showOnScreen = showOnScreen {
     attach(owner);
   }
@@ -1663,7 +1663,7 @@ class SemanticsNode extends AbstractNode with DiagnosticableTreeMixin {
   // bits are reserved for engine generated IDs.
   static const int _maxFrameworkAccessibilityIdentifier = (1<<16) - 1;

-  static int _lastIdentifier = 0;
+  static int _lastIdentifier = 123;
   static int _generateNewId() {
     _lastIdentifier = (_lastIdentifier + 1) % _maxFrameworkAccessibilityIdentifier;
     return _lastIdentifier;
```

</details>

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: shell/platform/common/accessibility_bridge_unittests.cc [hunk 1]
```diff
@@ -66,7 +66,74 @@ TEST(AccessibilityBridgeTest, basicTest) {
...
+
+  std::vector<int32_t> children{456, 789};
+  FlutterSemanticsNode root = CreateSemanticsNode(123, "root", &children);
+  FlutterSemanticsNode child1 = CreateSemanticsNode(456, "child 1");
+  FlutterSemanticsNode child2 = CreateSemanticsNode(789, "child 2");
+
+  bridge->AddFlutterSemanticsNodeUpdate(&child2);
+  bridge->AddFlutterSemanticsNodeUpdate(&root);
+  bridge->AddFlutterSemanticsNodeUpdate(&child1);
+  bridge->CommitUpdates();
```
Comment (loic-sharma) [inline shell/platform/common/accessibility_bridge_unittests.cc:118]: `CommitUpdates` must send updates to the `AXTree` in tree order, and, it must set the `AXTree`'s root. It sorts updates by creating sub-lists, each of which are in tree order. This test results in two sub-lists:

1. child 2
2. root, child 1

**Review by yaakovschectman — APPROVED** [review]
LGTM, just wondering about the one constant

File: shell/platform/common/accessibility_bridge.h [hunk 2]
```diff
@@ -50,7 +50,9 @@ class AccessibilityBridge
 
   //-----------------------------------------------------------------------------
   /// @brief      The ID of the root node in the accessibility tree. In Flutter,
-  //              this is always 0.
+  ///             this is always 0.
+  //  TODO(loicsharma): Remove this as it is incorrect in a multi-view world.
+  //  See: https://github.com/flutter/flutter/issues/119391
   static constexpr int32_t kRootNodeId = 0;
```
[the remark this answers is not in the document]
Comment (loic-sharma) [reply shell/platform/common/accessibility_bridge.h:56]: Yup, the other embedded will need to be updated before I can remove this.

**Review by goderbauer — APPROVED** [review]
LGTM

Thank you!

File: shell/platform/common/accessibility_bridge.cc [hunk 3]
```diff
@@ -95,6 +96,14 @@ void AccessibilityBridge::CommitUpdates() {
     }
   }
 
+  // Set the tree's root if necessary. The new root is the last list's first
+  // node.
+  if (!results.empty() && GetRootAsAXNode()->id() == ui::AXNode::kInvalidAXID) {
```
Comment (chunhtai) [inline shell/platform/common/accessibility_bridge.cc:101]: what if the root node changes? or would it never change?

File: shell/platform/common/accessibility_bridge.cc [hunk 4]
```diff
@@ -78,7 +78,8 @@ void AccessibilityBridge::CommitUpdates() {
   // entire subtree into a list. We pick another node from the remaining update,
   // and keep doing so until the update map is empty. We then concatenate the
   // lists in the reversed order, this guarantees parent updates always come
-  // before child updates.
+  // before child updates. Furthermore, the root is guaranteed to be the
```
[the remark this answers is not in the document]
Comment (loic-sharma) [reply shell/platform/common/accessibility_bridge.cc:81]: Ah this comment is incorrect, I'll fix it. It should be "if the root is in the update, ...".

The root node can be part of the update as the root node be modified. However, the tree's root node cannot change from one node to another.

File: shell/platform/common/accessibility_bridge.cc (hunk 3 above)
Comment (loic-sharma) [reply shell/platform/common/accessibility_bridge.cc:101]: A view's root node never changes once it is created. I can tweak this comment to make this more clear

File: shell/platform/common/accessibility_bridge.cc (hunk 4 above)
[the remark this answers is not in the document]
Comment (loic-sharma) [reply shell/platform/common/accessibility_bridge.cc]: I tweaked this comment, let me know what you think!

File: shell/platform/common/accessibility_bridge.cc (hunk 3 above)
Comment (loic-sharma) [reply shell/platform/common/accessibility_bridge.cc:102]: I tweaked this comment, let me know what you think!

**Review by chunhtai — APPROVED** [review]
LGTM

**Pull request merged**